flume-ng-channels/flume-redis-channel/src/main/java/org/apache/flume/channels/redis/RedisChannel.java

```diff
@@ -0,0 +1,462 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.flume.channels.redis;
+
+import org.apache.avro.io.BinaryDecoder;
+import org.apache.avro.io.BinaryEncoder;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.io.EncoderFactory;
+import org.apache.avro.specific.SpecificDatumReader;
+import org.apache.avro.specific.SpecificDatumWriter;
+import org.apache.flume.ChannelException;
+import org.apache.flume.Context;
+import org.apache.flume.Event;
+import org.apache.flume.FlumeException;
+import org.apache.flume.channel.BasicChannelSemantics;
+import org.apache.flume.channel.BasicTransactionSemantics;
+import org.apache.flume.channels.redis.tools.RedisInit;
+import org.apache.flume.channels.redis.tools.RedisOperator;
+import org.apache.flume.conf.ConfigurationException;
+
+import static org.apache.flume.channels.redis.RedisChannelConfiguration.*;
+
+import org.apache.flume.event.EventBuilder;
+import org.apache.flume.source.avro.AvroFlumeEvent;
+import org.apache.commons.lang.StringUtils;
+import org.jboss.netty.util.internal.StringUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import redis.clients.jedis.JedisPoolConfig;
+
+import com.google.common.net.HostAndPort;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class RedisChannel extends BasicChannelSemantics {
+
+  private static final Logger LOGGER = LoggerFactory
+      .getLogger(RedisChannel.class);
+
+  private final Properties redisConf = new Properties();
+  private final String channelUUID = UUID.randomUUID().toString();
+
+  private boolean parseAsFlumeEvent = DEFAULT_PARSE_AS_FLUME_EVENT;
+  private String queue_key;
+  private RedisOperator redisOperator;
+  private JedisPoolConfig jedisPoolConfig;
+  private RedisChannelCounter counter;
+
+  private final ThreadLocal<TransEvents> transFailoverController = new ThreadLocal<TransEvents>() {
+    @Override
+    public TransEvents initialValue() {
+      return new TransEvents(channelUUID);
+    }
+
+  };
+
+  private class TransEvents {
+    final String uuid;
+    final LinkedList<Event> failedEvents = new LinkedList<Event>();
+
+    TransEvents(String uuid) {
+      this.uuid = uuid;
+    }
+  }
+
+  @Override
+  public void start() {
+    try {
+      LOGGER.info("Starting Redis Channel: " + getName());
+      redisOperator = RedisInit.getInstance(redisConf, jedisPoolConfig);
+      counter.start();
+      super.start();
+    } catch (Exception e) {
```

Comment:
I have noticed a few occurrences of `catch (Exception e)`. According to https://stackoverflow.com/a/2416334/5323166 we should catch the most specific type we can. Can we catch more specific exceptions here?